### PR TITLE
Add support for `indexmap`

### PR DESCRIPTION
**Pull request by jgrund**
Add optional support for `indexmap` to

- `sets_duplicate_value_is_error`
- `maps_duplicate_key_is_error`
- `sets_last_value_wins`

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

File: serde_with/Cargo.toml [hunk 1]
```diff
@@ -41,6 +42,7 @@ rustversion = "1.0.0"
 serde = {version = "1.0.122", features = ["derive"]}
 serde_json = {version = "1.0.1", optional = true}
 serde_with_macros = {path = "../serde_with_macros", version = "1.5.0", optional = true}
+indexmap_crate = {package = "indexmap", version = "1.8", optional = true}
```
Comment (jonasbb) [inline serde_with/Cargo.toml:45]: nit: The dependencies are written in sorted order. Not a problem, though.

**Comment by jonasbb** [comment]
Hi, thanks for the contribution. I gladly merge `indexmap` support.
Do you also care about the `serde_as` transformations? These two links should show what I mean:
* https://docs.rs/serde_with/1.12.1/serde_with/guide/index.html#1-a-more-flexible-and-composable-replacement-for-the-with-annotation-called-serde_as-v150
* https://docs.rs/serde_with/1.12.1/serde_with/guide/serde_as_transformations/index.html#maps-to-vec-of-tuples

You could replace `#[serde(with = "indexmap::serde_seq")]` with `#[serde_as(as = "Vec<(_, _)>"]`, except that you can also control how the key/values are serialized. The matching implementations are in `src/{de,ser}/impls.rs`. No problem if you cannot add it.

**Comment by jgrund** [comment]
> You could replace #[serde(with = "indexmap::serde_seq")] with #[serde_as(as = "Vec<(_, _)>"], except that you can also control how the key/values are serialized. The matching implementations are in src/{de,ser}/impls.rs. No problem if you cannot add it.

Can I try this in followup PR?

**Comment by jonasbb** [comment]
> Can I try this in followup PR?

Sure, follow-ups welcome :) I will merge this but wait a little bit before releasing a new version.

bors merge

**Pull request merged**